### PR TITLE
Fix bugs in baro noise simulation

### PR DESCRIPTION
**Pull request by LorenzMeier**
Fixes https://github.com/PX4/Firmware/issues/9713

### CODE REVIEW

File: src/gazebo_mavlink_interface.cpp [hunk 1]
```diff
@@ -516,19 +516,28 @@ void GazeboMavlinkInterface::ImuCallback(ImuPtr& imu_message) {
...
+    static double y2;
+    static bool use_last = false;
+    if (!use_last) {
+      do {
+	x1 = 2.0 * ((double)rand() / (double)RAND_MAX) - 1.0;
+	x2 = 2.0 * ((double)rand() / (double)RAND_MAX) - 1.0;
+	w = x1 * x1 + x2 * x2;
+      } while ( w >= 1.0 );
+      w = sqrt( (-2.0 * log( w ) ) / w );
+      // calculate two values - the second value can be used next time becasue it is uncorrelated
```
Comment (LorenzMeier) [inline src/gazebo_mavlink_interface.cpp]: Typo: becasue

File: src/gazebo_mavlink_interface.cpp [hunk 2]
```diff
@@ -516,19 +516,28 @@ void GazeboMavlinkInterface::ImuCallback(ImuPtr& imu_message) {
...
-    do {
-     x1 = 2.0 * (rand() * (1.0 / (double)RAND_MAX)) - 1.0;
-     x2 = 2.0 * (rand() * (1.0 / (double)RAND_MAX)) - 1.0;
-     w = x1 * x1 + x2 * x2;
-    } while ( w >= 1.0 );
-    w = sqrt( (-2.0 * log( w ) ) / w );
-    y1 = x1 * w;
-    y2 = x2 * w;
+    double x1, x2, w, y1;
+    static double y2;
```
Comment (bkueng) [inline src/gazebo_mavlink_interface.cpp]: Can you avoid `static` here and make these a class attribute?

Comment (priseborough) [reply src/gazebo_mavlink_interface.cpp]: Fixed

**Comment by priseborough** [comment]
Does anyone know why the Gazebo 9 builds are failing on travis?

**Comment by TSC21** [comment]
It's not a build problem but rather a brew problem with the dependencies of Gazebo 9. Looking into it.

**Review by TSC21 — APPROVED** [review]

**Comment by TSC21** [comment]
Possible solution coming: https://github.com/Homebrew/homebrew-core/pull/29620

**Pull request merged**